### PR TITLE
Register `[[.rust_class` just as `$.rust_class`

### PR DESCRIPTION
**Pull request by shrektan**
Hi,

I believe it's necessary to register `[[` S3 method for exported Rust Class as well. 

The reason is that, only so, we can call the method dynamically, e.g., `method <- "method_abc"; RustClass[[method]]()`. 

This is not possible for `$`.

Thanks.

### CODE REVIEW

**Comment by yutannihilation** [comment]
Sounds good to me.

**Comment by andy-thomason** [comment]
This is great. Good to update the Change log, too.

We may export trait methods to s3 methods in the future in which case we may optionally overload `[[` as `Index`, but we
can cross that bridge when we come to it.

**Comment by yutannihilation** [comment]
Thanks for updating. One more thing, could you add a small test to verify `x[["foo"]]()` works as the same as `x$foo()` around here?

https://github.com/extendr/extendr/blob/a59d65c50e001810dadba2d8624517a36b4725d2/tests/extendrtests/tests/testthat/test-classes.R#L5

**Comment by shrektan** [comment]
@yutannihilation added.

**Review by yutannihilation — APPROVED** [review]
Looks good, thanks! I'll wait for a while before merging to see if there's any concern from other maintainers.

**Comment by yutannihilation** [comment]
I'm merging now. Thanks for the contribution!

**Pull request merged**
